sentry/src/main/java/io/sentry/rrweb/RRWebVideoEvent.java

```diff
@@ -0,0 +1,433 @@
+  public static final String REPLAY_CONTAINER = "mp4";
+  public static final String REPLAY_FRAME_RATE_TYPE_CONSTANT = "constant";
+  public static final String REPLAY_FRAME_RATE_TYPE_VARIABLE = "variable";
+
+  private @NotNull String tag;
+  private int segmentId;
+  private long size;
+  private int duration;
+  private @NotNull String encoding = REPLAY_ENCODING;
+  private @NotNull String container = REPLAY_CONTAINER;
+  private int height;
+  private int width;
+  private int frameCount;
+  private @NotNull String frameRateType = REPLAY_FRAME_RATE_TYPE_CONSTANT;
+  private int frameRate;
+  private int left;
+  private int top;
+  // to support unknown json attributes with nesting, we have to have unknown map for each of the
+  // nested object in json: { ..., "data": { ..., "payload": { ... } } }
+  private @Nullable Map<String, Object> unknown;
+  private @Nullable Map<String, Object> payloadUnknown;
+  private @Nullable Map<String, Object> dataUnknown;
+
+  public RRWebVideoEvent() {
+    super(RRWebEventType.Custom);
+    tag = EVENT_TAG;
+  }
+
+  @NotNull
+  public String getTag() {
+    return tag;
+  }
+
+  public void setTag(final @NotNull String tag) {
+    this.tag = tag;
+  }
+
+  public int getSegmentId() {
+    return segmentId;
+  }
+
+  public void setSegmentId(final int segmentId) {
+    this.segmentId = segmentId;
+  }
+
+  public long getSize() {
+    return size;
+  }
+
+  public void setSize(final long size) {
+    this.size = size;
+  }
+
+  public int getDuration() {
+    return duration;
+  }
+
+  public void setDuration(final int duration) {
+    this.duration = duration;
+  }
+
+  @NotNull
+  public String getEncoding() {
+    return encoding;
+  }
+
+  public void setEncoding(final @NotNull String encoding) {
+    this.encoding = encoding;
+  }
+
+  @NotNull
+  public String getContainer() {
+    return container;
+  }
+
+  public void setContainer(final @NotNull String container) {
+    this.container = container;
+  }
+
+  public int getHeight() {
+    return height;
+  }
+
+  public void setHeight(final int height) {
+    this.height = height;
+  }
+
+  public int getWidth() {
+    return width;
+  }
+
+  public void setWidth(final int width) {
+    this.width = width;
+  }
+
+  public int getFrameCount() {
+    return frameCount;
+  }
+
+  public void setFrameCount(final int frameCount) {
+    this.frameCount = frameCount;
+  }
+
+  @NotNull
+  public String getFrameRateType() {
+    return frameRateType;
+  }
+
+  public void setFrameRateType(final @NotNull String frameRateType) {
+    this.frameRateType = frameRateType;
+  }
+
+  public int getFrameRate() {
+    return frameRate;
+  }
+
+  public void setFrameRate(final int frameRate) {
+    this.frameRate = frameRate;
+  }
+
+  public int getLeft() {
+    return left;
+  }
+
+  public void setLeft(final int left) {
+    this.left = left;
+  }
+
+  public int getTop() {
+    return top;
+  }
+
+  public void setTop(final int top) {
+    this.top = top;
+  }
```

Comment:
l: Can you see this ever being any other value than zero?

Reply:
yeah, probalby when picture-in-picture mode or window is like a dialog window. And it is also relevant for flutter https://github.com/getsentry/rfcs/pull/129#discussion_r1483928742